src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/Endpoints/Diagnostics.cs

```diff
@@ -35,10 +36,11 @@ internal partial class RazorCustomMessageTarget
 
         var csharpTask = Task.Run(() => GetVirtualDocumentPullDiagnosticsAsync<CSharpVirtualDocumentSnapshot>(hostDocument, request.Identifier.Version, request.Identifier.TextDocumentIdentifier, request.CorrelationId, RazorLSPConstants.RazorCSharpLanguageServerName, cancellationToken), cancellationToken);
         var htmlTask = Task.Run(() => GetVirtualDocumentPullDiagnosticsAsync<HtmlVirtualDocumentSnapshot>(hostDocument, request.Identifier.Version, request.Identifier.TextDocumentIdentifier, request.CorrelationId, RazorLSPConstants.HtmlLanguageServerName, cancellationToken), cancellationToken);
+        var csharpAdditionalTask = Task.Run(() => GetCSharpAdditionalDiagnosticsAsync(hostDocument, request.CorrelationId, cancellationToken), cancellationToken);
 
         try
         {
-            await Task.WhenAll(htmlTask, csharpTask).ConfigureAwait(false);
+            await Task.WhenAll(htmlTask, csharpTask, csharpAdditionalTask).ConfigureAwait(false);
```

Comment:
I think `csharpAdditionalTask` will have to wait until after `csharpTask` is finished, or the analyzer might not have raised any diagnostics yet.

In fact, if we end up needing a custom endpoint for the additional diagnostics, we could just use that endpoint for both and call it once.